ebpf/profiling/Continuous.proto

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+syntax = "proto3";
+
+package skywalking.v3;
+
+option java_multiple_files = true;
+option java_package = "org.apache.skywalking.apm.network.ebpf.profiling.v3";
+option go_package = "skywalking.apache.org/repo/goapi/collect/ebpf/profiling/v3";
+
+import "common/Command.proto";
+
+service ContinuousProfilingService {
+    // query continuous profiling policy
+    rpc queryPolicies(ContinuousProfilingPolicyQuery) returns (Commands) {
+    }
+
+    // report the profiling task when reach the threshold of policy
+    rpc reportProfilingTask(ContinuousProfilingReport) returns (Commands) {
+    }
+}
+
+message ContinuousProfilingPolicyQuery {
+    // current agent contains service and policies
+    repeated ContinuousProfilingServicePolicyQuery policies = 1;
+}
+
+message ContinuousProfilingServicePolicyQuery {
+    // service name of the each process
+    string serviceName = 1;
+    // policy uuid
+    string uuid = 2;
+}
+
+message ContinuousProfilingReport {
+    // over threshold process entity
+    string layer = 1;
+    string serviceName = 2;
+    string instanceName = 3;
+    string processName = 4;
+
+    // reached thresholds
+    repeated ContinuousProfilingThreshold reaches = 5;
+
+    // target profiling task execute duration
+    int32 duration = 6;
+    // target profiling task
+    oneof targetTask {
+        ContinuousOnCPUProfilingTask onCPU = 7;
+        ContinuousOffCPUProfilingTask offCPU = 8;
+        ContinuousNetworkProfilingTask networkCPU = 9;
+    }
```

Comment:
Why do we need these three and network profiling even with some settings? Who pushed these network profiling settings?

Reply:
These three types of data are all reported by the rover. for the network profiling, the settings are set by the rover. It should perform sampling analysis on specifically set URIs from the policy.

Reply:
If those settings are static configurations on the rover side, why do we need to report them? We have collected data no matter what the sampling rules are.

Reply:
If you want to report metrics to show values that matched the policy(s), it is better you use meter channel to do that in an official way. The auto-profiling should be an event when triggered.

Reply:
> If those settings are static configurations on the rover side, why do we need to report them? We have collected data no matter what the sampling rules are.

It is not static. The sampling settings should be decided by which URI(or Regex) matches the threshold.

Reply:
I mean these
```
message ContinuousNetworkProfilingTaskSampling {
    string uriRegex = 1;
    int32 minDuration = 2;
    bool when4xx = 3;
    bool when5xx = 4;
    bool completeRequest = 5;
    bool completeResponse = 6;
}
```

I can't see these are as auto profiling parameters. They should be static, right?

Reply:
> If you want to report metrics to show values that matched the policy(s), it is better you use meter channel to do that in an official way. The auto-profiling should be an event when triggered.

Yes, the metrics would be using the native meter protocol. The cases only work for taking a snapshot for building an event on the OAP side.

Reply:
> I can't see these are as auto profiling parameters. They should be static, right?

Ok, the `uriRegex` should be reported, and other fields should be removed in this protocol.

Reply:
I think you should decouple the two things. 
1. Auto profiling triggered, which should include causes. Only URI cause may include URIRegex. But basically, this should only refer to an existing policy. Do we have a policy ID?
2. Other profiling data should be reported with a task(UUID or something) right? 

Reporting causes repeatedly seems duplicate and not helpful

Reply:
> Auto profiling triggered, which should include causes. Only URI cause may include URIRegex. But basically, this should only refer to an existing policy. Do we have a policy ID?

For now, we haven't the policy ID. If we add all policies with an ID may be too complex for this feature. Maybe I could copy the policy content into the report protocol? 

> Other profiling data should be reported with a task(UUID or something) right?

If the policy is reached, then the rover will call the report protocol to get a TaskID(which generates on the OAP). The report should be ignored when profiling or matching the silence policy.

Reply:
> For now, we haven't the policy ID. If we add all policies with an ID may be too complex for this feature. Maybe I could copy the policy content into the report protocol?

If you don't. I can't see why we plan to link the profiling to the policy. The policy could change with time. The link will lost the point.

Reply:
> For now, we haven't the policy ID. If we add all policies with an ID may be too complex for this feature. Maybe I could copy the policy content into the report protocol?

If you don't. I can't see why we plan to link the profiling to the policy. The policy could change with time. The link will lose the point.